components/CreateForm.js

```diff
@@ -1,57 +1,118 @@
 import { useState, useRef } from "react";
 import styled from "styled-components";
 import { Button, ButtonContainer } from "./Buttons";
+import { TbCheck, TbPlus } from "react-icons/tb";
+import { IconContext } from "react-icons";
+import { nanoid, customAlphabet } from "nanoid";
 
 function CreateForm() {
-  const [steps, setSteps] = useState(1);
   const buttonRef = useRef();
+  const [inputSteps, setInputSteps] = useState([
+    { step: 1, stepTitle: "", stepUrl: "", stepDescription: "" },
+  ]);
+
+  function handleFormChange(index, event) {
+    const data = [...inputSteps];
+    data[index][event.target.name] = event.target.value;
+    setInputSteps(data);
+  }
 
   function scrollToButton() {
     setTimeout(() => {
       buttonRef.current.scrollIntoView({ behavior: "smooth" });
     });
   }
-
   function handleAddStep() {
-    setSteps((prevSteps) => prevSteps + 1);
+    const additionalStep = {
+      step: inputSteps.length + 1,
+      stepTitle: "",
+      stepUrl: "",
+      stepDescription: "",
+    };
+    setInputSteps((prevInputSteps) => [...prevInputSteps, additionalStep]);
     scrollToButton();
   }
 
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+    const slugSuffix = customAlphabet(
+      "23456789abcdefghklmnpqrstuvwxyzABCDEFGHKLMNPQRSTUVWXYZ",
+      4
+    );
```

Comment:
All functions and constants that don't depend on state or props can be outside of components. If they are inside a component they will be reinitiated every time the component renders. (Only makes sense if you don't want them to be reinitiated of course)